packages/ia-components/sandbox/audio-players/youtube-player.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+// No restrictions for using iframe :)
+// functions and parameters such as autoplay, playVideo(), loadVideoById() won't work in all mobile environments
+
+class YoutubePlayer extends Component {
+
+  constructor(props) {
+    super(props);
+
+    const { selectedTrack, id } = props
+
+    this.state = {
+      playerAnchor: React.createRef(),
+      selectedTrack,
+      id
+    }
+
+    // use static methods?
+    this.loadAPI = this.loadAPI.bind(this);
+    this.loadPlayer = this.loadPlayer.bind(this);
+    this.onPlayerStateChange = this.onPlayerStateChange.bind(this);
+    this.onPlayerReady = this.onPlayerReady.bind(this);
+    this.onPlayerError = this.onPlayerError.bind(this);
+  }
+
+  // load api and iframe object
+  componentDidMount() {
+    this.loadAPI();
+    this.loadPlayer();
+  }
+
+  // Re render component only if track number changes
+  shouldComponentUpdate(nextProps) {
+    const { id } = this.state;
+    if (id !== nextProps.id) {
+      this.setState({
+        id: nextProps.id,
+        selectedTrack: nextProps.selectedTrack
+      });
+      return true;
+    }
+    return false;
```

Comment:
Rather than have two return statements in this method, we could simply `return id !== nextProps.id`, which I would do by way of a local `const` assignment to ensure the conditional code is updatable in one location. Something like: `const track_changed = id !== nextProps.id`